portfolio/src/main/webapp/script.js

```diff
@@ -51,10 +51,20 @@ function showPics() {
 
 /* Gets array list of messages from servlet, converts to JSON, and displays it on home page. */
 function getJSONMessages() {
-  fetch('/data').then(response => response.json()).then((messages) => {
-    const msgElement = document.getElementById('message-container');
-    msgElement.innerHTML = '';
-    msgElement.innerText = messages[0] + '\n' + messages[1] + '\n' + messages[2];
+  fetch('/data').then(response => response.json()).then((list) => {
+    const listElement = document.getElementById('list-container');
+    listElement.innerHTML = '';
+    var bucketList = "";
+
+    if (list.length === 0) {
+      bucketList = "";
+    } else {
+      for (i = 0; i < list.length; i++) {
```

Comment:
If list.length is 0, this for loop won't modify bucketList. Therefore I think it can be simplified to
```
let bucketList = "";
for (i = 0; i < list.length; i++) {
  // append strings
}
listElement.innerText = bucketList;
```